frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx

```diff
@@ -121,39 +225,341 @@ export const UninstallOperatorModal: React.FC<UninstallOperatorModalProps> = ({
-            })}
-          </p>
-        )}
-        {uninstallMessage && (
+        {!isSubmitFinished ? (
           <>
-            <h2>{t('olm~Message from Operator developer')}</h2>
-            <p>{uninstallMessage}</p>
+            <Instructions
+              name={name}
+              namespace={namespace}
+              removePlugins={removePlugins}
+              enabledPlugins={enabledPlugins}
+              uninstallMessage={uninstallMessage}
+            />
+            <OperandsSection
+              operands={operands}
+              operandsLoaded={operandsLoaded}
+              operandsLoadedErrorMessage={operandsLoadedErrorMessage}
+              csvName={csvName}
+              cancel={cancel}
+              isDeleteAllOperandsSelected={isDeleteAllOperandsSelected}
+              setIsDeleteAllOperandsSelected={setIsDeleteAllOperandsSelected}
+            />
           </>
+        ) : (
+          <SubmitResults
+            operatorUninstallErrorMessage={operatorUninstallErrorMessage}
+            name={name}
+            namespace={namespace}
+            isDeleteAllOperandsSelected={isDeleteAllOperandsSelected}
+            operandDeletionErrors={operandDeletionErrors}
+            csvName={csvName}
+            cancel={cancel}
+          />
         )}
       </ModalBody>
       <ModalSubmitFooter
-        inProgress={inProgress}
-        errorMessage={errorMessage}
+        inProgress={isSubmitInProgress}
         cancel={cancel}
-        submitDanger
-        submitText={t('olm~Uninstall')}
+        submitDanger={!isSubmitFinished} // if submit finished show a non-danger 'OK'
+        submitText={t(isSubmitFinished ? 'olm~OK' : 'olm~Uninstall')}
       />
     </form>
   );
 };
 
+const Instructions: React.FC<InstructionProps> = ({
+  name,
+  namespace,
+  removePlugins,
+  enabledPlugins,
+  uninstallMessage,
+}) => {
+  const { t } = useTranslation();
+  return (
+    <>
+      <p>
+        <Trans t={t} ns="olm">
+          Operator <strong>{{ name }}</strong> will be removed from <strong>{{ namespace }}</strong>
+          . Click on the checkbox below to also remove all Operands associated with this Operator.
+          If your Operator configured off-cluster resources, these will continue to run and require
+          manual cleanup.
+        </Trans>
+      </p>
+      {removePlugins && (
+        <p>
+          {t('olm~The console plugin provided by this operator will be disabled and removed.', {
+            count: enabledPlugins.length,
+          })}
+        </p>
+      )}
+      {uninstallMessage && (
+        <>
+          <h2>{t('olm~Message from Operator developer')}</h2>
+          <p>{uninstallMessage}</p>
+        </>
+      )}
+    </>
+  );
+};
+
+const OperandsSection: React.FC<OperandsSectionProps> = ({
+  operandsLoaded,
+  operands,
+  operandsLoadedErrorMessage,
+  csvName,
+  cancel,
+  isDeleteAllOperandsSelected,
+  setIsDeleteAllOperandsSelected,
```

Comment:
I'm on the fence about pulling these out to separate components since they take so many props. They're tightly coupled to the parent component, so it's probably better to inline them.

Reply:
arg!  I kinda had it that way initially but @TheRealJon asked for them to be broken out :-)

Reply:
I think it makes sense when you have simpler APIs that don't depend on the state of the parent. When you have to pass so much state and even callbacks through, they're really tightly coupled... Maybe you can break some of these up into smaller components with fewer props.

Reply:
@spadgett @dtaylor113 I would be fine with either inlining or further decoupling these as long as we don't declare them as "nested" React components inside the parent. IIRC, that's where my original suggestion came from. It might help to think about how this component might be used as a starting point for a more generalize bulk action modal in the future and create the helper components with that mind.

Reply:
Hi @TheRealJon, PTAL I refactored and moved a few of these inside the parent component, but none that are `React.FC` -those are still all outside of the parent component.